DiscordChatExporter.Core/Exporting/ExportContext.cs

```diff
@@ -48,6 +48,28 @@ var channel in Discord.GetGuildChannelsAsync(Request.Guild.Id, cancellationToken
         }
     }
 
+    public async ValueTask PopulateThreadsAsync(
+        Snowflake? id,
+        CancellationToken cancellationToken = default
+    )
+    {
+        if (id == null)
+        {
+            return;
+        }
+
+        Snowflake sid = (Snowflake)id;
+
+        if (_channelsById.ContainsKey(sid))
+        {
+            return;
+        }
+
+        var channel = await Discord.GetChannelAsync(sid, cancellationToken);
+
+        _channelsById[sid] = channel;
+    }
```

Comment:
Check for `null` at callsite instead

```suggestion
    public async ValueTask PopulateThreadAsync(
        Snowflake id,
        CancellationToken cancellationToken = default
    )
    {
        if (_channelsById.ContainsKey(id))
            return;

        var channel = await Discord.GetChannelAsync(id, cancellationToken);

        _channelsById[id] = channel;
    }
```